src/app/profiles.rs

```diff
@@ -206,13 +230,15 @@ where
                         State::Backoff(Delay::new(clock::now() + self.backoff))
                     }
                 },
-                State::Streaming(ref mut s) => match Self::proxy_stream(s, &mut self.tx) {
-                    Async::NotReady => return Ok(Async::NotReady),
-                    Async::Ready(StreamState::SendLost) => return Ok(().into()),
-                    Async::Ready(StreamState::RecvDone) => {
-                        State::Backoff(Delay::new(clock::now() + self.backoff))
+                State::Streaming(ref mut s) => {
+                    match Self::proxy_stream(s, &mut self.tx, &mut self.hangup) {
```

Comment:
It took me a moment to figure out why the whole match had "changed" in the diff", but I realised that adding `&mut self.hangup` to the call to `proxy_stream` had made the line too long, triggering `rustfmt` to reformat this block. So that's fine actually.